pkg/appmanager/validateResources.go

```diff
@@ -123,7 +123,7 @@ func (appMgr *Manager) checkValidIngress(
 					appMgr.resources.Delete(key, rsName)
 				}
 			}
-			appMgr.outputConfig()
+			appMgr.outputConfigLocked()
```

Comment:
As we chatted about, maybe it'd be better to only output the config if you had to delete a resource from the appMgr (to avoid writing out a new config if an Ingress we don't care about is created)